cmd/cli/wasm/wasm_run.go

```diff
@@ -139,111 +135,146 @@ func newRunCmd() *cobra.Command {
 func runWasm(cmd *cobra.Command, args []string, opts *WasmRunOptions) error {
 	ctx := cmd.Context()
 
-	j, err := CreateJob(ctx, args, opts)
+	job, err := CreateJobWasm(ctx, args, opts)
 	if err != nil {
 		return fmt.Errorf("creating job: %w", err)
 	}
 
-	if err := legacy_job.VerifyJob(ctx, j); err != nil {
-		return fmt.Errorf("verifying job: %w", err)
+	// Normalize and validate the job spec
+	job.Normalize()
+	if err := job.ValidateSubmission(); err != nil {
+		return fmt.Errorf("%s: %w", userstrings.JobSpecBad, err)
 	}
 
+	// TODO(forrest) [refactor]: this options is _almost_ useful. At present it marshals the entire
+	// job spec to yaml, said spec cannot be used with `bacalhau job run` since it contains fields that
+	// users are not permitted to set, like ID, Version, ModifyTime, State, etc.
+	// The solution here is to have a "JobSubmission" type that is different from the actual job spec.
 	if opts.RunTimeSettings.DryRun {
 		// Converting job to yaml
 		var yamlBytes []byte
-		yamlBytes, err = yaml.Marshal(j)
+		yamlBytes, err = yaml.Marshal(job)
 		if err != nil {
 			return fmt.Errorf("converting job to yaml: %w", err)
 		}
 		cmd.Print(string(yamlBytes))
 		return nil
 	}
 
-	executingJob, err := util.ExecuteJob(ctx, j, opts.RunTimeSettings)
+	api := util.GetAPIClientV2(cmd)
+	resp, err := api.Jobs().Put(ctx, &apimodels.PutJobRequest{Job: job})
 	if err != nil {
-		return fmt.Errorf("executing job: %w", err)
+		return fmt.Errorf("failed to submit job: %w", err)
+	}
+
+	if len(resp.Warnings) > 0 {
+		printWarnings(cmd, resp.Warnings)
+	}
+
+	if err := printer.PrintJobExecution(ctx, resp.JobID, cmd, &opts.RunTimeSettings.RunTimeSettings, api); err != nil {
+		return fmt.Errorf("failed to print job execution: %w", err)
 	}
 
-	return printer.PrintJobExecutionLegacy(ctx, executingJob, cmd, opts.DownloadSettings, opts.RunTimeSettings, util.GetAPIClient(ctx))
+	return nil
+}
+
+// TODO(forrest) [refactor]: dedupe from docker_run
+func printWarnings(cmd *cobra.Command, warnings []string) {
+	cmd.Println("Warnings:")
+	for _, warning := range warnings {
+		cmd.Printf("\t* %s\n", warning)
+	}
 }
 
-func CreateJob(ctx context.Context, cmdArgs []string, opts *WasmRunOptions) (*model.Job, error) {
+func CreateJobWasm(ctx context.Context, cmdArgs []string, opts *WasmRunOptions) (*models.Job, error) {
 	parameters := cmdArgs[1:]
 
 	entryModule, err := parseWasmEntryModule(ctx, cmdArgs[0])
 	if err != nil {
 		return nil, err
 	}
 
-	outputs, err := parse.JobOutputs(ctx, opts.SpecSettings.OutputVolumes)
+	envvar, err := parse.StringSliceToMap(opts.SpecSettings.EnvVar)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("wasm env vars invalid: %w", err)
 	}
-
-	nodeSelectorRequirements, err := parse.NodeSelector(opts.SpecSettings.Selector)
+	engineSpec, err := models.WasmSpecBuilder(&models.InputSource{
+		Source: entryModule,
+		Alias:  "TODO",
+		Target: "TODO",
```

Comment:
Unsure what `Alias` and `Target` actually do - obviously need to fix this, just not sure what the right answer is. cc @wdbaruni